docs/ota-client-guide/modules/ROOT/pages/garage-sign.adoc

```diff
@@ -0,0 +1,256 @@
+----
+
+The above commands create a local repository in a directory called `/tuf/<reponame>`. This directory will contain your `credentials.zip` file and has a tree structure as shown below:
+
+[source, bash]
+----
+tuf/<reponame>
+├── config.json
+├── credentials.zip
+├── keys
+│   ├── targets.pub <1>
+│   └── targets.sec <2>
+└── roles
+    ├── root.json   <3>
+    ├── targets.json.checksum
+    └── unsigned
+        └── targets.json <4>
+----
+
+<1> The public part of your `targets` key
+<2> The private part of your `targets` key
+<3> The `root.json` metadata file
+<4> The `targets.json` metadata file
+
+
+== Use cases of `garage-sign` with OTA Connect
+
+In most use cases, more than one `garage-sign` command is used to perform a specific task. A few use cases will be discussed in this section, however the use of the `garage-sign` tool is not limited to these.
+
+=== Generate new keys
+
+You might generate new keys if you:
+
+* xref:rotating-signing-keys.adoc#_rotate_the_online_keys_with_your_new_offline_keys[want to rotate your online keys with your new offline keys].
+* believe your keys have been compromised.
+* would like to revoke the current keys for signing metadata.
+
+==== Workflow to generate new keys
+
+When generating new keys for metadata, you perform the following steps:
+
+. First, generate the new `root` and `targets` keys by running the command `garage-sign key generate` twice.
++
+[source,bash]
+----
+garage-sign key generate \
+  --repo <reponame> \
+  --name myroot \ <1>
+  --type <ed25519|rsa> <2>
+garage-sign key generate \
+  --repo <reponame> \
+  --name mytargets \
+  --type <ed25519|rsa>
+----
++
+<1> You can give any name to the keys you generate, which helps to easily identify between the new `root` and `targets` keys. Here, the keys generated are given the names `myroot` and `mytargets`.
+<2> The key type must be specified as either Ed25519 or RSA. In OTA Connect, an RSA key is used for signing uptane metadata.
+
+. Afterwards, pull the latest `targets.json` file from OTA Connect with the command `garage-sign targets pull`.
++
+[source,bash]
+----
+garage-sign targets pull --repo <reponame>
+----
+
+. Next, use the command `garage-sign move-offline`. This command deletes the old `root` key from OTA Connect, downloads and signs the new `root` key with the newly generated keys, and pushes the new `root.json` file to OTA Connect.
++
+[source,bash]
+----
+garage-sign move-offline \
+  --repo <reponame> \
+  --old-root-alias originroot \ <1>
+  --new-root myroot \ <2>
+  --new-targets mytargets <3>
+----
++
+<1> This is an example of an alias given to the old `root` key. It gets saved with this name.
+<2> This is an example of the name given to your new `root` key which you previously generated using the command `garage-sign key generate`.
+<3> This is an example of the name given to your new `targets` key which you previously generated using the command `garage-sign key generate`.

```

Comment:
I like the idea of annotating these commands, since they are complex. However, I'm concerned that it makes copy-pasting the commands less user-friendly. (Or does Antora handle that more nicely than GitHub's rendering?)